### PR TITLE
Remove unsupported Window.dialogArguments

### PR DESCRIPTION
**Pull request by saschanaz**
#### Related issues

#6854

### CODE REVIEW

File: api/Window.json [hunk 1]
```diff
@@ -1694,54 +1694,6 @@
           }
         }
       },
-      "dialogArguments": {
```
Comment (Mouvedia) [inline api/Window.json:1697]: IE, Safari and Firefox had `window.dialogArguments`

**Comment by Elchi3** [comment]
I opened https://github.com/mdn/content/pull/10694 for Window.setCursor removal.

File: api/Window.json (hunk 1 above)
Comment (Mouvedia) [reply api/Window.json:1697]: _If_ the removal of `showModalDialog` was accompanied by the removal of `dialogArguments`, yes probably.
But like I said Firefox supported `dialogArguments` hence the table would be erroneous.
Someone with access to a browser farm will have to test it out.

**Comment by Mouvedia** [comment]
If `dialogArguments` gets fixed, follow it up with a revert of #10449.

**Comment by foolip** [comment]
Only the `dialogArguments` removal remains here now. I'll update the title.

**Comment by foolip** [comment]
I've dug into `dialogArguments`. Here's an example `showModalDialog()` exercising both `dialogArguments` and `returnValue`:

main.html:
```html
<!doctype html>
<button>open dialog</button>
<script>
document.querySelector('button').onclick = function() {
  var value = window.showModalDialog('dialog.html', ['hello', 'world']);
  alert('returned value: ' + value);
};
</script>
```

dialog.html:
```html
<!doctype html>
<div></div>
<script>
document.querySelector('div').textContent = JSON.stringify(window.dialogArguments);
</script>
<button>close me</button>
<script>
document.querySelector('button').onclick = function() {
  window.returnValue = 42;
  window.close();
};
</script>
```

Running this in Safari 15.4 shows that it still works, both `dialogArguments` and `returnValue`.

However, these are not normal properties of window that always exist, rather the `dialogArguments` property is set on the window object opened by `showModalDialog()`, and the `returnValue` property is read from the window object when it closes. The code for it is here:

https://github.com/WebKit/WebKit/blob/main/Source/WebCore/bindings/js/JSDOMWindowCustom.cpp#L466-L489

I think we should either have MDN pages + compat data for both `dialogArguments` and `returnValue`, or none of them. In this case, I think the pragmatic thing is to remove https://developer.mozilla.org/en-US/docs/Web/API/Window/dialogArguments and document everything in https://developer.mozilla.org/en-US/docs/Web/API/Window/showModalDialog instead. @Elchi3 WDYT?

**Comment by Elchi3** [comment]
> In this case, I think the pragmatic thing is to remove https://developer.mozilla.org/en-US/docs/Web/API/Window/dialogArguments and document everything in https://developer.mozilla.org/en-US/docs/Web/API/Window/showModalDialog instead. @Elchi3 WDYT?

Sounds good to me!

**Review by queengooborg — APPROVED** [review]
The MDN content updates have been merged now, so this PR is ready to go!

**Pull request merged**